src/utilities/link/types.ts

```diff
@@ -8,6 +8,8 @@ export interface LinkLikeComponentProps
   external?: boolean;
   /** Makes the browser download the url instead of opening it. Provides a hint for the downloaded filename if it is a string value. */
   download?: string | boolean;
+  /** Descriptive text to be read to screenreaders */
+  accessibilityLabel?: string;
```

Comment:
LinkLikeComponentProps / UnstyledLink are a funky corners of our setup, as they act as the interface for customizing how client-side routing gets handled. Because of that I think we want to stick as as close to behaving like a boring old `<a>` tags as possible, as evidenced by the props extending from `React.HTMLProps<HTMLAnchorElement>`

Because of that, in this one case of `UnstyledLink` I think we should stick to the api that already exists and we should use `aria-label`.